### PR TITLE
AEA-1837 Add pagination to pre-sign-send-page

### PR DESCRIPTION
**Pull request by booshibooshi**

### CODE REVIEW

File: tool/site/client-react/src/index.tsx [hunk 1]
```diff
@@ -20,27 +21,29 @@ async function startApplication(baseUrl: string): Promise<void> {
...
-            </Route>
-          </Switch>
-        </BrowserRouter>
-      </PageContainer>
+      <CookiesProvider>
+        <PageContainer>
+          <BrowserRouter>
+            <Switch>
+              <Route path={`${baseUrl}prescribe/edit`}>
+                <SendPreSignPage prescriptionId={urlParams.get("prescription_id")}/>
```
Comment (booshibooshi) [inline tool/site/client-react/src/index.tsx:29]: LoadPage?

File: tool/site/client-react/src/index.tsx [hunk 2]
```diff
@@ -20,27 +21,29 @@ async function startApplication(baseUrl: string): Promise<void> {
...
+              </Route>
+              <Route path={`${baseUrl}prescribe/send`}>
+                <SendPostSignPage token={urlParams.get("token")}/>
```
Comment (booshibooshi) [inline tool/site/client-react/src/index.tsx:32]: SendPage

File: tool/site/client-react/src/index.tsx [hunk 3]
```diff
@@ -20,27 +21,29 @@ async function startApplication(baseUrl: string): Promise<void> {
...
+              </Route>
+              <Route path={`${baseUrl}dispense/dispense`}>
+                <DispensePage prescriptionId={urlParams.get("prescription_id")}/>
+              </Route>
+              <Route path={`${baseUrl}dispense/claim`}>
+                <ClaimPage prescriptionId={urlParams.get("prescription_id")}/>
+              </Route>
+              <Route path={`${baseUrl}search`}>
```
Comment (callumtipper-IW) [inline tool/site/client-react/src/index.tsx:40]: the other paths are /search

File: tool/site/client-react/src/pages/sendPreSignPage.tsx [hunk 4]
```diff
@@ -19,6 +22,33 @@ const SendPreSignPage: React.FC<SendPreSignPageProps> = ({
...
+  /* Pagination ------------------------------------------------ */
+  const [addedListener, setAddedListener] = useState(false)
+  const [cookies] = useCookies()
+  const LEFT_ARROW_KEY = 37
+  const RIGHT_ARROW_KEY = 39
+  const handleKeyDown = useCallback((e: any) => {
+    if (e.keyCode === LEFT_ARROW_KEY) {
+      const previousPrescriptionId = cookies["Previous-Prescription-Id"]
+      if (previousPrescriptionId) {
+        customWindow.location = `${baseUrl}prescribe/edit?prescription_id=${encodeURIComponent(previousPrescriptionId)}`
```
Comment (mfjarvis) [inline tool/site/client-react/src/pages/sendPreSignPage.tsx:35]: Can we use `tool/site/client-react/src/browser/navigation.ts#redirect()` for redirects? That way we can mock it out for testing (js-dom doesn't support navigation)

**Review by callumtipper-IW — APPROVED** [review]

File: tool/site/client-react/src/pages/sendPreSignPage.tsx [hunk 5]
```diff
@@ -19,6 +22,33 @@ const SendPreSignPage: React.FC<SendPreSignPageProps> = ({
...
+        customWindow.location = `${baseUrl}prescribe/edit?prescription_id=${encodeURIComponent(nextPrescriptionId)}`
+      }
+    }
+  }, [baseUrl, cookies])
+  useEffect(() => {
+    if (!addedListener) {
+      document.addEventListener("keydown", handleKeyDown)
+    }
+    setAddedListener(true)
+  }, [addedListener, handleKeyDown])
```
Comment (mfjarvis) [inline tool/site/client-react/src/pages/sendPreSignPage.tsx:49]: Do we need to remove this listener at some point?

**Pull request merged**